exercises/book-store/canonical-data.json

```diff
@@ -13,99 +13,127 @@
         {
          "property": "total",
          "description": "Only a single book",
-         "basket": [1],
-         "targetgrouping": [[1]],
+         "input": {
+           "basket": [1],
+           "targetgrouping": [[1]]
```

Comment:
I strongly believe that targetgrouping is not an input and therefore it should not be placed in the `input` object.

I am assuming that inputs present in this `input` object are to be provided to the code under test. However, if we were to do this, then the code under test can simply read the `targetgrouping` to determine what the answer should be, instead of solving the problem of determining the optimal grouping.

I see the following in https://github.com/exercism/problem-specifications/blob/master/exercises/book-store/description.md

> Your mission is to write a piece of code to calculate the price of any conceivable shopping basket (containing only books of the same series), giving as big a discount as possible.

Well unfortunately for me, this neither confirms nor denies what I said. It doesn't explicitly say whether the code is responsible for calculating the expected grouping or whether the expected grouping will be provided. So, please decide this.

1. Confirm that the grouping will be provided to the code under test. In that case, I request:
  * The `basket` can simply be removed, it's redundant.
  * Shall there be cases that test what happens when non-optimal groupings are provided as input? For example, `"targetgrouping": [[1], [2]]`?
2. Remove `targetgrouping` from this file completely.
3. Move `targetgrouping` to `expected`, not `input`. TBD whether tests should request the target grouping instead of just the price (the description does not ask to determine the grouping!)

My preferences:

Remove completely > move to `expected` > further discussion to discover other options > leave it in `input`

Reply:
Strongly agree that `targetgrouping` should not be in `input`. As I understand it, it's a hint to test writers, but neither `input` nor `expected`. 

If valid in the new schema, I'd leave it where it is, as an attribute on the individual cases. If the schema prevents this, I'd favor moving it to a `comment` on each case. This could expose it to students, but that's not actually a problem; it shows very clearly the structure their code is expected to compute from the inputs.

Reply:
I agree that `targetgrouping` is not an input, but it isn't `expected` either, so I was hoping for others to weigh in. 👍   I think moving it to a comment is the best suggestion I have read so far.  Second best would be to remove completely.

Reply:
> If valid in the new schema

Currently uncertain. I could not think of a use for additional properties as of https://github.com/exercism/problem-specifications/pull/998#discussion_r151069698 but I see that this could give us one. So, we could allow additional properties on the top level. Or, `comment` will be fine.

OK, in that case for me:

Move to `comment` > Remove completely > leave it where it is > move to `expected` > further discussion to discover other options > leave it in `input`